ibm/data_source_ibm_is_security_group_rule_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAccIBMIsSecurityGroupRuleDataSourceBasic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckIBMIsSecurityGroupRuleDataSourceConfigBasic(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "id"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "security_group_id"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "id"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "direction"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "href"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "ip_version"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "protocol"),
+					resource.TestCheckResourceAttrSet("data.ibm_is_security_group_rule.is_security_group_rule", "remote.#"),
+				),
+			},
+		},
+	})
+}
+
+// Add the required existing resource details.
+func testAccCheckIBMIsSecurityGroupRuleDataSourceConfigBasic() string {
+	return fmt.Sprintf(`
+		data "ibm_is_security_group_rule" "is_security_group_rule" {
+			security_group_id = "r134-4d5db1a7-ca45-4c29-80d1-2a90cf94d11a"
```

Comment:
Do not hard code IDs in test case config

Reply:
done, added interpolation.
			security_group_id = ibm_is_security_group.testacc_security_group.id
			id = ibm_is_security_group_rule.testacc_security_group_rule_all.id